client/src/features/project/components/cloudStorage/AddOrEditCloudStorage.tsx

```diff
@@ -608,6 +565,81 @@ function SecretOptionWarning({
     </>
   );
 }
+interface OptionDataListProps {
+  examples: CloudStorageSchemaOptionExample[];
+  listId: string;
+}
+function OptionDataList({ examples, listId }: OptionDataListProps) {
+  if (!examples?.length) return null;
+
+  return (
+    <datalist id={listId}>
+      {examples?.map((e) => (
+        <option key={`${listId}__${e.value}`} value={e.value} />
+      ))}
+    </datalist>
+  );
+}
+
+interface TruncatedTextProps {
+  collapsedLines?: number;
+  linesTolerance?: number;
+  text: string;
+}
+function TruncatedText({
+  collapsedLines = 2,
+  linesTolerance = 1,
+  text,
+}: TruncatedTextProps) {
+  const [isTruncated, setIsTruncated] = useState(true);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const toggleTruncate = () => {
+    setIsTruncated(!isTruncated);
+  };
```

Comment:
Prefer `useCallback()` version:

```suggestion
  const toggleTruncate = useCallback(() => {
    setIsTruncated((isTruncated) => !isTruncated);
  }, []);
```